CONTRIBUTING.md

```diff
@@ -92,9 +92,37 @@ provide any backward compatible guarantees.
 
 #### Backporting
 
-We only accept backports of critical bugs, security issues, or bugs without easy
-workarounds, any backport MUST not be breaking for either API or behavioral
-changes. We generally do not accept PRs against older release branches.
+We generally do not accept PRs directly against release branches, while we might
+accept backports of fixes/changes already merged into the main branch.
+
+We generally allow backports of following changes to all supported branches:
+
+- Critical bugs fixes, security issue fixes, or fixes for bugs without easy
+workarounds.
+- Dependency bumps for CVE (usually limited to CVE resolution; backports of
+non-CVE related version bumps are considered exceptions to be evaluated case by
+case)
+- Changes required to support new Kubernetes versions, when possible.
+- Changes to use the latest Go patch version to build controller images.
+- Changes to bump the Go minor version used to build controller images, if the
+Go minor version of a supported branch goes out of support (e.g. to pick up
+bug and CVE fixes). This has no impact on users importing Cluster API Provider
+Metal3 as we won't modify the version in go.mod and the version in the Makefile
+does not affect them.
+
+We generally allow backports of following changes only to the latest supported
+branch:
+
+- Improvements to existing docs
```

Comment:
We want updates to docs to `main`, not latest release branch. I don't think we have put any doc updates (besides triggers and release related updates) to release branches. `main` is not supported branch IMO, it is development branch.

Reply:
I think this could be ok. We havent done yet doesnt mean we wouldnt. And if some bug fix would require to change the documentation , then this would anyway stay true.

Reply:
I'd assume bugfix would include the documentation change in the same PR anyways?

Reply:
As discussed offline, lets just merge this list to the above list.